src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+
+import StorageUtils from '../../../utils/storage';
+import {Context} from '../../../context';
+import {checkUserBlock, getOtherProfile} from '../../../service/profile';
+import {setOtherProfileFeed} from '../../../context/actions/otherProfileFeed';
+
+const useOtherProfileScreenHooks = (targetUserProfileId: string, username: string) => {
+  /**
+   * State
+   */
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [isProfileFetching, setIsProfileFetching] = React.useState<boolean>(true);
+  const [isBlocked, setIsBlocked] = React.useState<boolean | undefined>(undefined);
+  const [isBlocking, setIsBlocking] = React.useState<boolean | undefined>(undefined);
+  const [otherProfileData, setOtherProfileData] = React.useState<any>({});
+  const [, setOtherProfileUserId] = React.useState<string>('');
+
+  /**
+   * Other Profile Context
+   */
+  const [otherProfileFeeds, dispatchOtherProfile] = (React.useContext(Context) as any)
+    ?.otherProfileFeed;
+  const {feeds} = otherProfileFeeds;
+
+  const getCache = async () => {
+    const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true';
+    const isBlockingFromCache =
+      StorageUtils.blockingStatus.getForKey(targetUserProfileId) === 'true';
+
+    try {
+      const otherProfileDataCache =
+        StorageUtils.otherProfileData.getForKey(targetUserProfileId) || '{}';
+      setOtherProfileData(JSON.parse(otherProfileDataCache));
+      if (otherProfileDataCache !== '{}') {
+        setIsProfileFetching(false);
+      }
+    } catch (e) {
+      console.log('error parsing JSON other profile data', e);
+    }
+
+    try {
+      const feedsCache = StorageUtils.otherProfileFeed.getForKey(targetUserProfileId) || '[]';
+
+      setOtherProfileFeed(JSON.parse(feedsCache), dispatchOtherProfile);
+    } catch (e) {
+      console.log('error parsing JSON other profile feed', e);
+    }
+
+    setIsBlocked(isBlockedFromCache);
+    setIsBlocking(isBlockingFromCache);
+  };
+
+  const fetchFeeds = async () => {
+    setIsLoading(true);
+    setIsLoading(false);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `fetchFeeds` function currently doesn't do anything. It sets `isLoading` to true and then immediately sets it to false. If this function is supposed to fetch feeds, it should have some logic to do so.



<!-- This is an auto-generated comment by OSS CodeRabbit -->